### PR TITLE
deploy Azure Container Registry Example

### PR DESCRIPTION
**Pull request by ExchMaster**
# Description

This example deploys a premium Azure Container Registry suitable for hosting docker containers.  The registry will be deployed to the MLZ shared services resource group using default naming unless alternative values are provided at run time.  

This update uses the bicep shared variable file pattern as described here: https://docs.microsoft.com/en-us/azure/azure-resource-manager/bicep/patterns-shared-variable-file

## Issue reference

This PR is related to the following issue: #517
I did attempt to retrieve MLZ outputs using a deployment object, but ultimately failed to successfully do so do the dynamic nature of how the outputs are parsed/discovered at run-time. More research is needed, but felt this pattern solved the majority of the issue and is worthy of doing now while we continue to investigate how to use deployment objects for this problem.

## Checklist

Please make sure you've completed the relevant tasks for this PR out of the following list:

* [see above] All acceptance criteria in the backlog item are met
* [x ] The documentation is updated to cover any new or changed features
* [x ] Manual tests have passed
* [x ] Relevant issues are linked to this PR

NOTE:  I am adding multiple examples quickly with over-lapping pull-requests so in an effort to avoid merge conflicts I was planning on updating the ./src/bicep/examples/README.md with all the information at once.   I did add an initial update in my first PR to that ReadMe updating the docs on creating the deployment Variables file.  I'll come back and update the index of examples after I update/add a few more of them.

### CODE REVIEW

File: src/bicep/mlz.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
     "_generator": {
       "name": "bicep",
       "version": "0.4.1008.15138",
-      "templateHash": "17151661725468237381"
+      "templateHash": "15169166681509362530"
```
Comment (brooke-hamilton) [inline src/bicep/mlz.json:8]: This file appears unchanged except for the `templateHash` value.

File: src/bicep/examples/deploymentVariables.json [hunk 2]
```diff
@@ -0,0 +1,77 @@
+{
```
Comment (brooke-hamilton) [inline src/bicep/examples/deploymentVariables.json:1]: Please remove this file as we are now excluding it in the `.gitignore`

File: src/bicep/examples/containerRegistry/README.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+### Template Parameters
+
+Template Parameters Name | Description
+-----------------------| -----------
+contRegistryName | The name of the Container Registry.  If not specified, the name will default to the MLZ default naming pattern.  
+targetResourceGroup | The name of the resource group where the Container Registry will be deployed.   If not specified, the resource group name will default to the shared services MLZ resource group name and subscription.
+
+### Generate MLZ VAriable File (deploymentVariables.json)
+
+For instructions on generating 'deploymentVariables.json' using both Azure PowerShell and Azure CLI, please see the [README at the root of the examples folder](..\README.md).
```
Comment (brooke-hamilton) [inline src/bicep/examples/containerRegistry/README.md:31]: The link is broken--change to a forward slash. 

Also, the README at that location does not contain instructions on generating the `deploymentVariables.json` file.

```suggestion
For instructions on generating 'deploymentVariables.json' using both Azure PowerShell and Azure CLI, please see the [README at the root of the examples folder](../README.md).
```

File: src/bicep/examples/containerRegistry/README.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+cd .\containerRegistry
+New-AzSubscriptionDeployment -DeploymentName deployContainerRegistry -TemplateFile .\contRegistry.bicep -Location 'eastus'
+```
+
+```Azure CLI
+az login
+cd src/bicep
+az deployment sub create -n contoso -f mlz.bicep -l eastus --parameters resourcePrefix=contoso
+cd examples
+az deployment sub show -n contoso --query properties.outputs >> ./deploymentVariables.json
```
Comment (brooke-hamilton) [inline src/bicep/examples/containerRegistry/README.md:54]:
```suggestion
az deployment sub show -n contoso --query properties.outputs > ./deploymentVariables.json
```

**Review by brooke-hamilton — CHANGES_REQUESTED** [review]

File: src/bicep/examples/deploymentVariables.json (hunk 2 above)
Comment (ExchMaster) [reply src/bicep/examples/deploymentVariables.json]: Done.

File: src/bicep/mlz.json (hunk 1 above)
Comment (ExchMaster) [reply src/bicep/mlz.json]: The templatehash value is calculated by bicep during bicep build, I just updated the feature branch from main to get all relevant changes which caused the templateHash to match up.  Should be good to go now.

File: src/bicep/examples/containerRegistry/README.md (hunk 3 above)
Comment (ExchMaster) [reply src/bicep/examples/containerRegistry/README.md]: The README.md in the root of the examples folder was updated as a part of a different PR.  Should be good to go now.

**Review by brooke-hamilton — APPROVED** [review]
@ExchMaster thanks for this! ❤️

**Pull request merged**